src/commands/release-asset/upload.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * @license
+ * Copyright 2025 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Command, Flags, Args } from '@oclif/core';
+import { getBalenaSdk, stripIndent, getVisuals } from '../../utils/lazy';
+import { commitOrIdArg } from '../release';
+import * as fs from 'fs/promises';
+import * as path from 'path';
+
+export default class ReleaseAssetUploadCmd extends Command {
+	public static description = stripIndent`
+		Upload a release asset.
+
+		Upload a file as a release asset with the specified key. If the asset already exists,
+		use the --overwrite flag to replace it. You can customize the upload behavior with
+		--chunk-size and --parallel-chunks options for larger files.
+	`;
+
+	public static examples = [
+		'$ balena release-asset upload 1234567 ./path/to/config.json --key config.json',
+		'$ balena release-asset upload a777f7345fe3d655c1c981aa642e5555 ./app.tar.gz --key app.tar.gz --overwrite',
+		'$ balena release-asset upload 1234567 ./file.bin --key large-file.bin --chunk-size 10485760 --parallel-chunks 10',
+	];
+
+	public static args = {
+		commitOrId: commitOrIdArg({
+			description: 'the commit or ID of the release',
```

Comment:
I keep forgetting, do we want to still allow IDs? If not then I assume the ones we currently allow are just remnants, and of course then we don't have to support that here cc @thgreasi